packages/providers/src/provider.ts

```diff
@@ -64,30 +64,32 @@ export type ContractResult = {
   bytecode: string;
 };
 
+type ConsensusParameters = {
+  contractMaxSize: BN;
+  maxInputs: BN;
+  maxOutputs: BN;
+  maxWitnesses: BN;
+  maxGasPerTx: BN;
+  maxScriptLength: BN;
+  maxScriptDataLength: BN;
+  maxStorageSlots: BN;
+  maxPredicateLength: BN;
+  maxPredicateDataLength: BN;
+  maxGasPerPredicate: BN;
+  gasPriceFactor: BN;
```

Comment:
I reckon that with the solution proposed here, we can also close https://github.com/FuelLabs/fuels-ts/issues/784